bin/lib/Logos/Generator/Base/Property.pm

```diff
@@ -27,47 +41,89 @@ sub definition {
 	my $propertyAssociationPolicy = $property->associationPolicy;
 	my $wrapValue = !($propertyAssociationPolicy =~ /RETAIN|COPY/);
 
-	# Build getter
-	my $getter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
-	$getter_func .= "static $propertyType $propertyGetterName($propertyClass * __unused self, SEL __unused _cmd) ";
-	if($wrapValue) {
-		$getter_func .= "{ NSValue * value = objc_getAssociatedObject(self, (void *)$propertyGetterName); $propertyType rawValue; [value getValue:&rawValue]; return rawValue; }";
-	} else {
-		$getter_func .= "{ return ($propertyType)objc_getAssociatedObject(self, (void *)$propertyGetterName); }";
-	}
+	if (!$readonly) {
+		# Build getter
+		my $getter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
+		$getter_func .= "static $propertyType $propertyGetterName($propertyClass * __unused self, SEL __unused _cmd) ";
+		if($wrapValue) {
+			$getter_func .= "{ NSValue * value = objc_getAssociatedObject(self, (void *)$propertyGetterName); $propertyType rawValue; [value getValue:&rawValue]; return rawValue; }";
+		} else {
+			$getter_func .= "{ return ($propertyType)objc_getAssociatedObject(self, (void *)$propertyGetterName); }";
+		}
 
-	# Build setter
-	my $setter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
-	$setter_func .= "static void $propertySetterName($propertyClass * __unused self, SEL __unused _cmd, $propertyType rawValue) ";
-	if($wrapValue) {
-		$setter_func .= "{ NSValue * value = [NSValue valueWithBytes:&rawValue objCType:\@encode($propertyType)]; objc_setAssociatedObject(self, (void *)$propertyGetterName, value, OBJC_ASSOCIATION_RETAIN_NONATOMIC); }";
-	} else {
-		$setter_func .= "{ objc_setAssociatedObject(self, (void *)$propertyGetterName, rawValue, $propertyAssociationPolicy); }";
+		# Build setter
+		my $setter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
+		$setter_func .= "static void $propertySetterName($propertyClass * __unused self, SEL __unused _cmd, $propertyType rawValue) ";
+		if($wrapValue) {
+			$setter_func .= "{ NSValue * value = [NSValue valueWithBytes:&rawValue objCType:\@encode($propertyType)]; objc_setAssociatedObject(self, (void *)$propertyGetterName, value, OBJC_ASSOCIATION_RETAIN_NONATOMIC); }";
+		} else {
+			$setter_func .= "{ objc_setAssociatedObject(self, (void *)$propertyGetterName, rawValue, $propertyAssociationPolicy); }";
+		}
+
+		return "$getter_func;\n$setter_func";
 	}
 
-	return "$getter_func; $setter_func";
+	return "";
```

Comment:
Could we have the `else` clause here and explain why we don't add getters/setters for readonly properties? (the it'd contain the `return "";`)
Am I right thinking that the idea is to let the user create a getter method, as opposed to public readonly in `@interface` and private readwrite in `@implementation` for the normal iOS development?

Reply:
> Am I right thinking that the idea is to let the user create a getter method

Yes